apps/webapp/app/routes/_app.orgs.$organizationSlug/route.tsx

```diff
@@ -1,63 +1,95 @@
+import { Breadcrumb, BreadcrumbLink } from "~/components/navigation/Breadcrumb";
+import { PageNavigationIndicator } from "~/components/navigation/PageNavigationIndicator";
+import { SideMenu } from "~/components/navigation/SideMenu";
+import { useOptionalOrganization } from "~/hooks/useOrganizations";
+import { useOptionalProject } from "~/hooks/useProject";
+import { useTypedMatchData } from "~/hooks/useTypedMatchData";
+import { useUser } from "~/hooks/useUser";
+import { OrganizationsPresenter } from "~/presenters/OrganizationsPresenter.server";
+import { getCurrentProjectId } from "~/services/currentProject.server";
+import { getImpersonationId } from "~/services/impersonation.server";
 import { requireUserId } from "~/services/session.server";
+import { telemetry } from "~/services/telemetry.server";
+import { Handle } from "~/utils/handle";
 import { organizationPath } from "~/utils/pathBuilder";
 
+const ParamsSchema = z.object({
+  organizationSlug: z.string(),
+  projectParam: z.string().optional(),
+});
+
 export const loader = async ({ request, params }: LoaderFunctionArgs) => {
   const userId = await requireUserId(request);
-  const { organizationSlug } = params;
-  invariant(organizationSlug, "organizationSlug not found");
+  const impersonationId = await getImpersonationId(request);
+
+  const { organizationSlug, projectParam } = ParamsSchema.parse(params);
 
-  const organization = await getOrganizationFromSlug({
+  const orgsPresenter = new OrganizationsPresenter();
+  const { organizations, organization, project } = await orgsPresenter.call({
     userId,
-    slug: organizationSlug,
+    request,
+    organizationSlug,
+    projectSlug: projectParam,
   });
 
-  if (organization === null) {
-    throw new Response("Not Found", { status: 404 });
-  }
-
   telemetry.organization.identify({ organization });
 
-  const session = await setCurrentOrg(organization.slug, request);
-
-  return typedjson(
-    {
-      organization,
-    },
-    {
-      headers: {
-        "Set-Cookie": await commitCurrentOrgSession(session),
-      },
-    }
-  );
+  return typedjson({
+    organizations,
+    organization,
+    currentProject: project,
+    isImpersonating: !!impersonationId,
+  });
+};
+
+export const handle: Handle = {
+  breadcrumb: (match) => {
+    const data = useTypedMatchData<typeof loader>(match);
+    return (
+      <BreadcrumbLink to={match.pathname} title={data?.organization.title ?? "Organization"} />
+    );
+  },
 };
 
 export default function Organization() {
+  const { organization, currentProject, organizations, isImpersonating } =
+    useTypedLoaderData<typeof loader>();
+  const user = useUser();
+
+  //the side menu won't change projects when using the switcher unless we use the hook (on project pages)
+  const project = useOptionalProject() ?? currentProject;
+
   return (
     <>
-      <Outlet />
+      <div className="grid grid-cols-[14rem_1fr] overflow-hidden">
```

Comment:
The new SideMenu lives at the Org level. Plus the breadcrumb and the new loading indicator.